### PR TITLE
fix: allows no rel attribute on external links in the nav

### PR DESCRIPTION
**Pull request by abea**
**Summary**

Currently it's not possible to specify that there should be no `rel` attribute on external nav links to use the default behavior. Setting to an empty string, as [indicated in the documentation](https://v1.vuepress.vuejs.org/theme/default-theme-config.html#navbar-links), is falsy, so you end up in the final ternary, which uses `noopener noreferrer` for external links.

This change allows devs to pass in a falsy, non-`undefined`, value to the `rel` property to have no rel attribute on the link. It also change the fallback of an empty string to `null` since an empty `rel` attribute has no particular meaning, so it's better to have it not there at all.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

I could not tell from the contributor guide or the package scripts how to build this for linking to a project and testing. I'd be happy to do that with some direction.
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/docs/docs/theme/default-theme-config.md [hunk 1]
```diff
@@ -69,14 +69,14 @@ module.exports = {
 }
 ```
 
-Outbound links automatically get `target="_blank" rel="noopener noreferrer"`. You can offer `target` and `rel` to customize the attributes:
+Outbound links automatically get `target="_blank" rel="noopener noreferrer"`. You can offer `target` and `rel` to customize the attributes. Passing a "falsy" value to the `rel` attribute will disable that attribute for a link:
```
Comment (abea) [inline packages/docs/docs/theme/default-theme-config.md:72]: Oh, maybe I should make this clearer to say that the falsy value can't be `undefined`.

Comment (newsbielt703) [reply packages/docs/docs/theme/default-theme-config.md]: IMO, change the rule. Let user set `false` rather than falsy value to disable the attribute.

Comment (abea) [reply packages/docs/docs/theme/default-theme-config.md]: That makes sense.

**Comment by abea** [comment]
Should I create an issue to update this in the Chinese docs or something?

File: packages/@vuepress/theme-default/components/NavLink.vue [hunk 2]
```diff
@@ -71,10 +71,13 @@ export default {
       if (this.isNonHttpURI) {
         return null
       }
+      if (!this.item.rel && this.item.rel === false) {
+        return null
+      }
```
Comment (newsbielt703) [inline packages/@vuepress/theme-default/components/NavLink.vue:76]:
```suggestion
      if (this.item.rel === false) {
        return null
      }
```

**Comment by newsbielt703** [comment]
> Should I create an issue to update this in the Chinese docs or something?

I'll take it later. No worries.

**Pull request merged**